### PR TITLE
Create separate plugin for magnetometer

### PR DESCRIPTION
**Pull request by EliaTarasov**
Move magnetometer calculation to separate plugin.

### CODE REVIEW

**Comment by EliaTarasov** [comment]
Logs:
1. [master](https://review.px4.io/plot_app?log=755d751e-9431-4281-8d4a-c1c6a425f0cf)
2. [pr](https://review.px4.io/plot_app?log=929c2791-fbb1-47df-90a3-b1ddfb7ea520)

**Comment by EliaTarasov** [comment]
@TSC21 Good enough?

**Comment by EliaTarasov** [comment]
@julianoes what can you tell? The same idea is PX4/sitl_gazebo#295

**Review by TSC21 — CHANGES_REQUESTED** [review]
You need to add this plugin to all models, not just to the Iris model.

File: CMakeLists.txt [hunk 1]
```diff
@@ -262,6 +262,7 @@ set(sensor_msgs
   msgs/Range.proto
   msgs/SITLGps.proto
   msgs/OpticalFlow.proto
+  msgs/Magnetometer.proto
```
Comment (TSC21) [inline CMakeLists.txt]: Magnetometer = sensor. Data = `magnetic_field`.

File: include/gazebo_magnetometer_plugin.h [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+
+#include <common.h>
+
+#include <geo_mag_declination.h>
+
+namespace gazebo {
+
+static constexpr auto kDefaultMagnetometerTopic = "mag";
+static constexpr auto kDefaultPubRate = 20.0; // [Hz]
+static constexpr auto kDefaultNoiseDensity = 0.01; // [gauss / sqrt(hz)]
```
Comment (TSC21) [inline include/gazebo_magnetometer_plugin.h]: Is this noise model correct for mag field data?

Comment (EliaTarasov) [reply include/gazebo_magnetometer_plugin.h]: I'm not sure. This is the default [model](https://github.com/PX4/sitl_gazebo/blob/master/src/gazebo_mavlink_interface.cpp#L613). Maybe we need approach similar to the `gazebo_imu_plugin` or default ROS sensor model.

File: CMakeLists.txt (hunk 1 above)
Comment (EliaTarasov) [reply CMakeLists.txt]: Done.

File: include/gazebo_magnetometer_plugin.h (hunk 2 above)
Comment (EliaTarasov) [reply include/gazebo_magnetometer_plugin.h]: Any suggestions on that?

Comment (EliaTarasov) [reply include/gazebo_magnetometer_plugin.h]: @priseborough can you take a quick look at this please?

**Comment by EliaTarasov** [comment]
1. [master](https://review.px4.io/plot_app?log=2a745041-c99c-4637-a92c-0edb5b645be1)
2. [pr](https://review.px4.io/plot_app?log=d806a895-ae5a-4cda-9e65-2216b41f9d9f)
3. [real](https://review.px4.io/plot_app?log=7d53f424-a146-4636-8566-debc8f8796bf)

Note: [Random Walk](https://github.com/PX4/sitl_gazebo/blob/8da1c9a984b8c10eb55da467c6b0738fddae4a97/include/gazebo_magnetometer_plugin.h#L72) is not specified in ADIS16448 datasheet directly.

Pr model looks much more noisy. 

@TSC21 Can you please review it?
Thanks!

**Comment by EliaTarasov** [comment]
Now magnetometer looks much more [smoothly](https://review.px4.io/plot_app?log=16e8df3b-c0e4-44e4-91af-ebca5fd777b2) and similar to the [real](https://review.px4.io/plot_app?log=7d53f424-a146-4636-8566-debc8f8796bf) one.

@TSC21 Any notes?

**Comment by EliaTarasov** [comment]
@TSC21 Upd

**Comment by TSC21** [comment]
@julianoes all good for you?

**Review by julianoes — APPROVED** [review]

**Review by TSC21 — APPROVED** [review]
LGTM! Thanks @EliaTarasov

**Pull request merged**